### PR TITLE
fix: bugfix for popup

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-5573 MF-5932 MF-5935 MF-5942 MF-5944 MF-5945 MF-5946 MF-5954 MF-5955

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/shared/src/hooks/usePersonaContext.ts [hunk 1]
```diff
@@ -134,6 +134,9 @@ function usePersonaContext(initialState?: {
         selectedAccount,
         setSelectedAccount,
         avatar,
+        refreshAvatar: () => {
```
Comment (guanbinrui) [inline packages/shared/src/hooks/usePersonaContext.ts:137]: refreshAvatar: refresh,

File: packages/mask/popups/pages/Wallet/WalletGuard/index.tsx [hunk 2]
```diff
@@ -28,7 +30,8 @@ export const WalletGuard = memo(function WalletGuard() {
             </>
         )
     }
-    if (hitPaymentPasswordGuard) {
+
+    if (hitPaymentPasswordGuard && !hasPassword) {
```
Comment (Jack-Works) [inline packages/mask/popups/pages/Wallet/WalletGuard/index.tsx]: this is a little bit strange. why not read hasPassword from API instead of using params?

Comment (nuanyang233) [reply packages/mask/popups/pages/Wallet/WalletGuard/index.tsx:34]: In some scenarios, we will directly open the Popup window wallet page. At this time, the result of the api has not been returned and will be false by default, causing it to be directly routed to the page for setting the password. 比如这个 bug

Comment (nuanyang233) [reply packages/mask/popups/pages/Wallet/WalletGuard/index.tsx]: It's to fix this [issue](https://mask.atlassian.net/browse/MF-5944). In this case it opens the wallet page of the popup window. But the default value stored in react-query is `false`. There is also no way to render loading status by `isLoading` or `isPending` before Dehydrate. So I added a params to fix it.

Comment (Jack-Works) [reply packages/mask/popups/pages/Wallet/WalletGuard/index.tsx]: can you use suspense loading of hasPassword to fix this problem?

File: packages/shared/src/hooks/usePersonaContext.ts (hunk 1 above)
Comment (nuanyang233) [reply packages/shared/src/hooks/usePersonaContext.ts:137]: This will cause the type annotation

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**